### DISCUSSION TITLE
Extension Mangement with ExtensionSettings

### DISCUSSION THREAD

**Discussion by FusionTerror** [mozilla/policy-templates#1009] [Q&A] [answered]
I'm having some issues with managing extensions in Firefox with the **latest** ADMX templates. Within Group Policy under _"Computer Configuration\Administrative Templates\Mozilla\Firefox\Extensions"_ I have _"Extension Management", "Extension Update", "Extensions to Install", "Extension to Uninstall", and "Prevent extensions from being disabled or removed"_. I'd like to configure Firefox to force install two extensions (uBlock and Adblock Plus) and allow other extensions I specify to be installed by the user, but not forced. Also, I'd like extensions to update themselves and not be disabled/removed by the user. Finally, I'd like all extensions not allowed, or force installed to be removed from the browser/system. My issue is that I created a JSON as shown in [ExtensionSettings](https://github.com/mozilla/policy-templates#extensionsettings), but it's not effecting the browser after forcing a gpupdate on the system. I'm not sure if the other policy settings are overriding it, or if I'm doing this wrong. I've attached my settings below and JSON for extension management.

Policy Settings:
Extension Management - Enabled (Options Value):
```json
{
    "*": {
        "install_mode": "blocked",
        "blocked_install_message": "The system administrator has disabled software installation."
    },
    "uBlock0@raymondhill.net": {
        "install_mode": "force_installed",
        "install_url": "https://addons.mozilla.org/firefox/downloads/file/4092158/ublock_origin-1.48.4.xpi"
    },
    "{d10d0bf8-f5b5-c8b4-a8b2-2b9879e08c5d}": {
        "install_mode": "force_installed",
        "install_url": "https://addons.mozilla.org/firefox/downloads/file/4080833/adblock_plus-3.16.2.xpi"
    },
    "{446900e4-71c2-419f-a6a7-df9c091e268b}": {
        "install_mode": "allowed"
    }
}
```
Extension Update - Enabled
Extensions to Uninstall - Enabled (Option Values): *
Prevent extensions from being disabled or removed - Enabled
Extensions to Install - Enabled (Option Values):
- https://addons.mozilla.org/firefox/downloads/file/4080833/adblock_plus-3.16.2.xpi
- https://addons.mozilla.org/firefox/downloads/file/4092158/ublock_origin-1.48.4.xpi

**Comment by mkaply** [comment]
ExtensionSettings completely replaces 

Extensions to Uninstall - Enabled (Option Values): *
Prevent extensions from being disabled or removed - Enabled
Extensions to Install - Enabled (Option Values):

So you shouldn't need to do that.

What do you see in about:policies?

From what I see, that extensionsettings should work. I'll doublecheck.

**Reply by FusionTerror** [reply]
about:policies contains the JSON object under ExtensionSettings, but it has no effect on the browser since the listed extensions to force install didn't install, and the allow list isn't able to be installed from.

**Comment by mkaply** [comment]
OK, let me do some testing.

**Comment by mkaply** [comment, marked as answer]
I found it.

install_mode

should be

installation_mode

See:

https://github.com/mozilla/policy-templates/blob/master/README.md#extensionsettings

and again, you shouldn't need to use any other settings, that should uninstall things as well.

**Reply by FusionTerror** [reply]
Thanks for taking a look for me, it seems the correction of my typo has solved the issue. I'm wondering if you knew how to set the startup page to [Google](https://google.com) since Mozilla/Firefox/Home page contains "Start Page" but doesn't allow for a custom link. I've got the home button set to something else, so I cannot use the home button on startup.

**Reply by mkaply** [reply]
There's not any way to do what you're asking as the homepage and the home button would point to the same page. We don't have a way to separate them.

**Comment by anewuser** [comment]
> I'd like to configure Firefox to force install two extensions (uBlock and Adblock Plus)

You should **never** install any other content blocker alongside uBlock Origin. [Read this explanation](https://twitter.com/gorhill/status/1033706103782170625).
